File: app/src/main/java/com/sd/www/http/ui/SyncRequestActivity.java
package com.sd.www.http.ui;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import com.sd.lib.http.IRequest;
import com.sd.lib.http.IResponse;
import com.sd.lib.http.impl.GetRequest;
import com.sd.www.http.R;

/**
 * 同步请求demo
 */
public class SyncRequestActivity extends AppCompatActivity
{
    private TextView mTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sync_request);
        mTextView = findViewById(R.id.tv_result);
    }

    public void onClickRequest(View view)
    {
        new Thread(new Runnable()
        {
            @Override
            public void run()
            {
                try
                {
                    IRequest request = new GetRequest();
                    //设置请求地址
                    request.setBaseUrl("https://www.baidu.com/");
                    //发起请求，得到Response对象
                    IResponse response = request.execute();
                    //请求结果以字符串返回
                    final String result = response.getAsString();

                    runOnUiThread(new Runnable()
                    {
                        @Override
                        public void run()
                        {
                            mTextView.setText(result);
                        }
                    });
                } catch (Exception e)
                {
                    e.printStackTrace();
                }
            }
        }).start();
    }
}

File: app/src/main/java/com/sd/www/http/activity/SyncRequestActivity.kt
package com.sd.www.http.activity

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.sd.lib.http.exception.HttpException
import com.sd.lib.http.exception.HttpExceptionCancellation
import com.sd.lib.http.exception.HttpExceptionResultIntercepted
import com.sd.lib.http.impl.GetRequest
import com.sd.www.http.databinding.ActivitySyncRequestBinding
import com.sd.www.http.model.WeatherModel
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.MainScope
import kotlinx.coroutines.cancel
import kotlinx.coroutines.launch

/**
 * 同步请求demo
 */
class SyncRequestActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var _binding: ActivitySyncRequestBinding

    val _mainScope: CoroutineScope = MainScope()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        _binding = ActivitySyncRequestBinding.inflate(layoutInflater)
        setContentView(_binding.root)
    }

    override fun onClick(v: View?) {
        if (v == _binding.btnStart) {
            requestData()
        }
    }

    private fun requestData() {
        _mainScope.launch {
            val request = GetRequest().apply {
                this.baseUrl = "http://www.weather.com.cn/data/cityinfo/101010100.html"
                this.extra = "SyncRequestActivity"
            }

            val result = request.parseSuspend(WeatherModel::class.java)
            if (result.isSuccess) {
                _binding.tvResult.text = result.data!!.weatherinfo!!.city
            } else {
                val desc = when (val exception = result.exception) {
                    is HttpExceptionCancellation -> "请求被取消"
                    is HttpExceptionResultIntercepted -> "请求结果被拦截"
                    is HttpException -> exception.getDescFormat(application)
                    else -> exception.toString()
                }
                _binding.tvResult.text = desc
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        _mainScope.cancel()
    }
}